src/main/plugins/org.dita.pdf2/xsl/fo/links.xsl

```diff
@@ -489,7 +489,7 @@ See the accompanying LICENSE file for applicable license.
       </xsl:choose>
     </xsl:template>
 
-  <xsl:template match="*[contains(@class,' topic/link ')]" mode="processLink">
+  <xsl:template match="*[contains(@class,' topic/link ')][not(empty(@href) or @href='')]" mode="processLink">
```

Comment:
I would like us to at some point consider adding empty `@href` removal to preprocessing. Then we would not have to have `@href = ''` tests in transtype specific code. Or is there any reason why to use an empty `@href` attribute? `URI.create("file:/base/topic.xml").resolve("")` will resolve to `file:/base/`, but I don't see any reason to allow users to link to _directories_. If we remove empty `@href`, you could still link to those with `href="./"`.

Reply:
:+1: for removing empty `@href` (and empty `@conref`) early in initial processing. It's caused problems several times over the year when other steps don't test for it, and try to open the directory -- exactly as you've described. For an author running a build it's extremely confusing and hard to track down, because it usually appears as "Cannot open `/base/`" in the error. So yeah -- leaving it in means a lot of messy checks in the code, and means if you forget the check, things go wrong.

The only reason I can think of for having it is if you have some element with required `@href` attribute, but you don't actually want to specify one -- maybe because you use `@conref` or `@keyref`. Technically that's what the `-dita-use-conref-target` token is for but most people don't know about that. Even for that case though -- the empty `@href` would only be needed for initial parsing / validation, and generic DITA processing code that follows won't look for it.